src/tasks/AppleAppBuilder/Xcode.cs

```diff
@@ -259,6 +259,16 @@ public void CreateXcodeProject(string projectName, string cmakeDirectoryPath)
     {
         // bundle everything as resources excluding native files
         var predefinedExcludes = new List<string> { ".dll.o", ".dll.s", ".dwarf", ".m", ".h", ".a", ".bc", "libmonosgen-2.0.dylib", "libcoreclr.dylib", "icudt*" };
+        if (useNativeAOTRuntime)
+        {
+            predefinedExcludes.Add(".dll");
+            predefinedExcludes.Add(".pdb");
+            predefinedExcludes.Add(".json");
+            predefinedExcludes.Add(".txt");
+            predefinedExcludes.Add(".bin");
+            predefinedExcludes.Add(".dSYM");
+        }
```

Comment:
Note to myself: these shouldn't be needed once we carefully construct the publish folder on Helix